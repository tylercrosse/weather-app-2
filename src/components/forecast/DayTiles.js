import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import 'moment-timezone';

export const DayTile = ({ day, timezone }) => {
  const time = moment.unix(day.time).tz(timezone).format('ddd D');
  return (
    <div className="dayTile">
      <div>{time}</div>
      <img src={`../../${day.icon}.svg`} alt={day.icon} />
      <div>
        {Math.round(day.temperatureMax)} / {Math.round(day.temperatureMin)}
      </div>
    </div>
  );
};

DayTile.propTypes = {
  day: PropTypes.shape({
    temperatureMax: PropTypes.number.isRequired,
    temperatureMin: PropTypes.number.isRequired,
    icon: PropTypes.string.isRequired,
  }),
  timezone: PropTypes.string.isRequired,
}

const DayTiles = ({ timezone, dailyData }) => {
  const dayTiles = dailyData
    .slice(0, 7)
    .map(day => <DayTile key={day.time} day={day} timezone={timezone} />);

  return (
    <div className="forecast__dayTiles">
      {dayTiles}
    </div>
  );
};

DayTiles.propTypes = {
  timezone: PropTypes.string.isRequired,
  dailyData: PropTypes.array.isRequired
}

export default DayTiles;
