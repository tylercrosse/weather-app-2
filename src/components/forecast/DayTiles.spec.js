import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import DayTiles, { DayTile } from './DayTiles';

const setupDayTiles = () => {
  const props = {
    timezone: 'America/Los_Angeles',
    dailyData: []
  }

  const component = <DayTiles {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<DayTiles />', () => {
  it('should render correctly', () => {
    const { wrapper } = setupDayTiles();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})

const setupDayTile = () => {
  const props = {
    day: {
      temperatureMax: 75,
      temperatureMin: 45,
      icon: 'fog',
    },
    timezone: 'America/Los_Angeles'
  }

  const component = <DayTile {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<DayTile />', () => {
  it('should render correctly', () => {
    const { wrapper } = setupDayTile();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
