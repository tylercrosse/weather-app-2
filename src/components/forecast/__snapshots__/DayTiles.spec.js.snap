// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayTile /> should render correctly 1`] = `
<div
  className="dayTile"
>
  <div>
    Invalid date
  </div>
  <img
    alt="fog"
    src="../../fog.svg"
  />
  <div>
    75
     / 
    45
  </div>
</div>
`;

exports[`<DayTiles /> should render correctly 1`] = `
<div
  className="forecast__dayTiles"
/>
`;
